Clarify initial route usage with cached engines. (#3551)
flutter/website: src/docs/development/add-to-app/android/add-flutter-fragment.md

By providing a pre-warmed `FlutterEngine`, as previously shown, your app renders the
first Flutter frame as quickly as possible.

#### Initial route with a cached engine

The concept of an initial route is available when configuring a 
`FlutterActivity` or a `FlutterFragment` with a new `FlutterEngine`. However,

> `FlutterActivity` and `FlutterFragment` do not offer the concept of an initial

Reviewer: Nit:
```suggestion
`FlutterActivity` and `FlutterFragment` don't offer the concept of an initial
```
One reason that such contractions are useful: it's sometimes easy for a reader to miss the word "not," whereas it's harder to misread "don't" as "do."

See https://developers.google.com/style/contractions

route when using a cached engine. This is because a cached engine is expected
to already be running Dart code, which means it's too late to configure the
initial route.

Developers that would like their cached engine to begin with a custom
initial route can configure their cached `FlutterEngine` to use a custom initial
route just before executing the Dart entrypoint. The following example

> demonstrates the use of an initial route with a cached engine.

Reviewer: Nit: 
```suggestion
demonstrates the use of an initial route with a cached engine:
```

  }
}
```
{% sample Kotlin %}
<?code-excerpt "MyApplication.kt" title?>
```kotlin
class MyApplication : Application() {
  lateinit var flutterEngine : FlutterEngine

  override fun onCreate() {
    super.onCreate()

    // Instantiate a FlutterEngine.
    flutterEngine = FlutterEngine(this)

    // Configure an initial route.
    flutterEngine.navigationChannel.initialRoute = "your/route/here";

    // Start executing Dart code to pre-warm the FlutterEngine.
    flutterEngine.dartExecutor.executeDartEntrypoint(
      DartExecutor.DartEntrypoint.createDefault()
    )

    // Cache the FlutterEngine to be used by FlutterActivity or FlutterFragment.
    FlutterEngineCache
      .getInstance()
      .put("my_engine_id", flutterEngine)
  }
}
```
{% endsamplecode %}

By setting the initial route of the navigation channel, the associated
`FlutterEngine` displays the desired route upon initial execution of the
`runApp()` Dart function.

Changing the initial route property of the navigation channel after the initial
execution of `runApp()` has no effect. Developers who would like to use
the same `FlutterEngine` between different `Activity`s and `Fragment`s and

> switch the route between those displays need to setup a method channel and

Reviewer: Nit:
```suggestion
switch the route between those displays need to set up a method channel and
```
